### PR TITLE
[WIP] [Arm64] Apply Span clear opt to spans with refs

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
I believe there is no reason this should not work.  However it is currently hitting #12549 during crossgen.

@dotnet/arm64-contrib @jashook @jkotas

**Comment by jkotas** [comment]
It may work today, but it is based on pretty fragile assumptions. And the GC crashes where object reference is not written atomically take months to trace down.

What kind of perf improvements do you see from this? I would expect it to regress the performance for small sizes. It is hard to tell whether it is a good trade-off - we do not have many real cases where this method is hit.

If assembly version of JIT_MemSet is better than C#, using it for the non-GC refs case should be fine - and it is likely more beneficial.

File: src/mscorlib/shared/System/Span.NonGeneric.cs [hunk 1]
```diff
@@ -483,6 +483,23 @@ internal static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSize
...
+            {
+                // Limit clear to 4096 bytes at a time to allow GC interruption
+                nuint clearLength = (byteLength > 4096) ? 4096 : byteLength;
+                Unsafe.InitBlockUnaligned(ref Unsafe.As<IntPtr, byte>(ref Unsafe.AddByteOffset<IntPtr>(ref ip, offset)), 0, (uint)clearLength);
+                offset += clearLength;
+                byteLength -= clearLength;
+            }
+#else
+            // TODO: Enable optimization on other platforms with JIT_MemSet GC refs atomicity guarantee
             // TODO: Perhaps do switch casing to improve small size perf
```
Comment (jkotas) [inline src/mscorlib/shared/System/Span.NonGeneric.cs]: You can delete this comment. Using `switch` to optimize this was proven to be a bad idea.

**Comment by sdmaclea** [comment]
@jkotas I have not been able to run the code since it causes an assert in crossgen.  I uploaded it for conversation.

I have not observed an interesting use case where this is hit.  I just noticed it could be done while I was looking for for the non-gc ref optimization.  (Code moved since it was introduced for AMD64)

I am abandoning the PR.

I will create another one to remove stray comment.

**Comment by jkotas** [comment]
Thanks anyway

**Pull request closed without merging**